iothub_client/samples/iothub_client_device_cbor_sample/iothub_client_device_cbor_sample.c

```diff
@@ -0,0 +1,956 @@
+                }
+            }
+        }
+        else
+        {
+            (void)printf("The device client did not receive expected system property content type value: %s.\n", CONTENT_TYPE_C2D);
+        }
+    }
+
+    (void)printf("\n"); // Formatting
+
+    return IOTHUBMESSAGE_ACCEPTED;
+}
+
+//
+// Encoding/Decoding with CBOR library
+//
+static bool parse_cbor_desired_property(DEVICE_TWIN_UPDATE_STATE update_state, const unsigned char* payload, Car* out_car)
+{
+    CborError rc; // CborNoError == 0
+    bool result;
+
+    CborParser parser;
+    CborValue root;
+    CborValue state_root;
+
+    CborValue change_oil_reminder;
+    CborValue max_speed;
+    CborValue software_version;
+
+    rc = cbor_parser_init(payload, strlen((char*)payload), 0, &parser, &root);
+    if (rc)
+    {
+        (void)printf("Failed to initiate parser: CborError %d.\n", rc);
+        exit(rc);
+    }
+
+    // Check if this is the full twin document or only the desired properties.
+    if (update_state == DEVICE_TWIN_UPDATE_COMPLETE) // full twin document (desried and reported)
+    {
+        rc = cbor_value_map_find_value(&root, twin_desired_property, &root);
+        if (rc)
+        {
+            (void)printf("Error when searching for %s: CborError %d.\n", twin_desired_property, rc);
+            exit(rc);
+        }
+    }
+
+    // change_oil_reminder
+    rc = cbor_value_map_find_value(&root, twin_property_change_oil_reminder_name, &change_oil_reminder);
+    if (rc)
+    {
+        (void)printf("Error when searching for %s: CborError %d.\n", twin_property_change_oil_reminder_name, rc);
+        exit(rc);
+    }
+    if (cbor_value_is_valid(&change_oil_reminder))
+    {
+        if (cbor_value_is_boolean(&change_oil_reminder))
+        {
+            rc = cbor_value_get_boolean(&change_oil_reminder, &out_car->change_oil_reminder);
+            if (rc)
+            {
+                (void)printf("Failed to get boolean value of property %s: CborError %d.\n", twin_property_change_oil_reminder_name, rc);
+                exit(rc);
+            }
+            else
+            {
+                (void)printf("Parsed desired property `%s`: %s\n", twin_property_change_oil_reminder_name, out_car->change_oil_reminder ? "true" : "false");
+                result = true;
+            }
+        }
+        else
+        {
+            (void)printf("`%s` property value was not a boolean.\n", twin_property_change_oil_reminder_name);
+            result = false;
+        }
+    }
+    else
+    {
+        (void)printf("`%s` property name was not found in desired property message.\n", twin_property_change_oil_reminder_name);
+        result = false;
+    }
+
+    // state
+    rc = cbor_value_map_find_value(&root, twin_property_state_name, &state_root);
+    if (rc)
+    {
+        (void)printf("Error when searching for %s: CborError %d.\n", twin_property_state_name, rc);
+        exit(rc);
```

Comment:
same here as mentioned above. I'm not so sure this should be `exit`ing

Reply:
see above note.